operator/standalone/delete_it_test.go

```diff
@@ -0,0 +1,288 @@
+////go:build integration
+
+package standalone
+
+import (
+	"context"
+	"fmt"
+	"github.com/vshn/appcat-service-postgresql/apis/postgresql/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+
+	pipeline "github.com/ccremer/go-command-pipeline"
+	helmv1beta1 "github.com/crossplane-contrib/provider-helm/apis/release/v1beta1"
+	"github.com/stretchr/testify/suite"
+	"github.com/vshn/appcat-service-postgresql/operator/operatortest"
+)
+
+type DeleteStandalonePipelineSuite struct {
+	operatortest.Suite
+}
+
+func TestDeleteStandalonePipeline(t *testing.T) {
+	suite.Run(t, new(DeleteStandalonePipelineSuite))
+}
+
+func (ts *DeleteStandalonePipelineSuite) BeforeTest(suiteName, testName string) {
+	ts.Context = pipeline.MutableContext(context.Background())
+	setClientInContext(ts.Context, ts.Client)
+	ts.RegisterScheme(helmv1beta1.SchemeBuilder.AddToScheme)
+}
+
+func (ts *DeleteStandalonePipelineSuite) Test_DeleteHelmRelease() {
+	tests := map[string]struct {
+		prepare                    func(releaseNameString string)
+		givenReleaseName           string
+		expectedHelmReleaseDeleted bool
+		expectedError              error
+	}{
+		"GivenNonExistingHelmRelease_WhenDeleting_ThenExpectNoFurtherAction": {
+			prepare:                    func(releaseName string) {},
+			givenReleaseName:           "postgresql-release",
+			expectedHelmReleaseDeleted: true,
+			expectedError:              nil,
+		},
+		"GivenAnExistingHelmRelease_WhenHelmReleaseStillExists_ThenExpectReconciliation": {
+			prepare: func(releaseName string) {
+				release := newPostgresqlHelmRelease(releaseName)
+				ts.EnsureResources(release)
+			},
+			givenReleaseName:           "postgresql-release",
+			expectedHelmReleaseDeleted: false,
+			expectedError:              nil,
+		},
+		"GivenAnExistingHelmRelease_WhenDeletingGeneratesAnError_ThenReturnError": {
+			prepare:                    func(releaseName string) {},
+			givenReleaseName:           "",
+			expectedHelmReleaseDeleted: false,
+			expectedError:              fmt.Errorf("resource name may not be empty"),
+		},
+	}
+	for name, tc := range tests {
+		ts.Run(name, func() {
+			d := &DeleteStandalonePipeline{
+				client: ts.Client,
+				instance: newBuilderInstance("instance", "namespace").
+					setDeploymentNamespace(tc.givenReleaseName).
+					get(),
+				helmReleaseDeleted: false,
+			}
+			tc.prepare(tc.givenReleaseName)
+			err := d.deleteHelmRelease(ts.Context)
+
+			// Assert
+			if tc.expectedError == nil {
+				ts.NotFoundResource(
+					types.NamespacedName{Name: tc.givenReleaseName, Namespace: tc.givenReleaseName},
+					&helmv1beta1.Release{},
+				)
+
+			}
```

Comment:
```suggestion
			if tc.expectedError != "" {
				ts.Assert().EqualError(err, tc.expectedError)
				return
			}
```

Reply:
Why would you remove the resource check?

Reply:
I check if the resource does not exist in case there's no error, not when there is an error

Reply:
Yeah, but on line 81 you do it again. So with your given code if there's no error you do the check twice.

Reply:
@ccremer 's comment is outdated. I updated the code and I check for no resource only in case there's no error

Reply:
Okay. Well the current code LGTM. You probably have to wait for Chris then to answer your question. As I don't know how it looked before.